### PR TITLE
[docs] Add applitools documentation

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 1]
```diff
@@ -1 +1,223 @@
...
+** `batchName` - The name of the batch (group of visual tests).
+** `baselineName` - The name of the baseline (visual test name).
+** `matchLevel` - The https://applitools.com/docs/common/cmn-eyes-match-levels.html#Available[match level].
+** `appName` - The name of the application under test.
+** `elementsToIgnore` - The comma separated list of locators of elements to ignorer.
+** `areasToIgnore` - The comma separated list of locators of page areas to ignorer.
+** `hostApp` - The application running the AUT (e.g. Safari, Chrome).
+** `hostOS` - The host OS running the AUT.
+** `viewportSize` - The required viewport size for the environment.
+** `baselineEnvName` - The name of the baseline's environment.
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-applitools.adoc:171]: it's potentially dangerous property to specify :)

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 2]
```diff
@@ -1 +1,223 @@
...
+|Default
+|Description
+
+|`web.visual.applitools.execute-api-key`
+|`empty`
+|Allows users to execute tests but does not allow them to access the tests and test results, change data through the API, or make any changes to the tests.
+
+|`web.visual.applitools.read-api-key`
+|`empty`
+|Allows users to access tests and test results, but does not allow them to execute tests, change data through the API, or make any changes to the tests.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-applitools.adoc:31]: we need more details how to get these keys

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 3]
```diff
@@ -1 +1,223 @@
...
+
+|`web.visual.applitools.match-level`
+|`EXACT`
+|See https://applitools.com/docs/common/cmn-eyes-match-levels.html#Available[available match levels].
+
+|`web.visual.applitools.server-uri`
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-applitools.adoc:37]: it should go first

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 4]
```diff
@@ -1 +1,223 @@
...
+|`https://eyesapi.applitools.com`
+|The URI of the rest server.
+
+|`web.visual.applitools.app-name`
+|`empty`
+|The name of the application under test.
+
+|===
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-applitools.adoc:45]: it's not clear which property is required and which is not

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 5]
```diff
@@ -1 +1,223 @@
...
+
+|Property Name
+|Default
+|Description
+
+|`web.visual.applitools.host-os`
+|`empty`
+|The host OS running the AUT.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-applitools.adoc:61]: `AUT` - not clear

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 6]
```diff
@@ -1 +1,223 @@
...
+----
+Given I am on a page with the URL 'https://example.com/'
+When I run visual test with Applitools using:
+|appName    |batchName      |baselineName |action                  |elementsToIgnore|
+|example.com|example-testing|main-app-page|CHECK_INEQUALITY_AGAINST|By.tagName(h1)  |
+----
+
+== Steps
+
+=== Visual check with default applitools configuration
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-applitools.adoc:130]:
```suggestion
=== Visual check with default Applitools configuration
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 7]
```diff
@@ -1 +1,223 @@
...
+** `elementsToIgnore` - The comma separated list of locators of elements to ignorer.
+** `areasToIgnore` - The comma separated list of locators of page areas to ignorer.
+** `hostApp` - The application running the AUT (e.g. Safari, Chrome).
+** `hostOS` - The host OS running the AUT.
+** `viewportSize` - The required viewport size for the environment.
+** `baselineEnvName` - The name of the baseline's environment.
+** `searchContext` - The locator of a search context
+** `readApiKey` - The key to execute tests.
+** `executeApiKey` - The key to access tests and test results.
+** `serverUri` - The URI of the rest server.
```
[the remark this answers is not in the document]
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-applitools.adoc:175]: this capability is provided by the plugin design

File: docs/modules/plugins/pages/plugin-applitools.adoc (hunk 1 above)
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-applitools.adoc:171]: agree :) but it's exposed to end users so we have to document this :)

File: docs/modules/plugins/pages/plugin-applitools.adoc [hunk 8]
```diff
@@ -1 +1,235 @@
...
+
+|`web.visual.applitools.match-level`
+|`EXACT`
+|See https://applitools.com/docs/common/cmn-eyes-match-levels.html#Available[available match levels].
+
+|`web.visual.applitools.app-name`
+|`empty`
+|The name of the application under test.
+
+|===
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-applitools.adoc:47]: Rename properties for Applitools visual plugin

```suggestion
|===

|Property Name
|Default
|Description

|`ui.visual.applitools.server-uri`
|`https://eyesapi.applitools.com`
|The URI of the rest server.

|`ui.visual.applitools.execute-api-key`
|`empty`
|Allows users to execute tests but does not allow them to access the tests and test results, change data through the API, or make any changes to the tests. For more details see how to https://applitools.com/docs/topics/overview/obtain-api-key.html[obtain API keys].

|[subs=+quotes]`*web.visual.applitools.read-api-key*`
|`empty`
|Allows users to access tests and test results, but does not allow them to execute tests, change data through the API, or make any changes to the tests. For more details see how to https://applitools.com/docs/topics/overview/obtain-api-key.html[obtain API keys].

|`ui.visual.applitools.match-level`
|`EXACT`
|See https://applitools.com/docs/common/cmn-eyes-match-levels.html#Available[available match levels].

|`ui.visual.applitools.app-name`
|`empty`
|The name of the application under test.

|===
```

File: docs/modules/plugins/pages/plugin-visual.adoc [hunk 9]
```diff
@@ -344,7 +338,7 @@ Establishes baseline or compares against existing one using the ignoring option
 When I $actionType baseline with `$name` ignoring:$ignoredElements using screenshot configuration:$screenshotConfiguration
 ----
 
-* `actionType` - {actions}
+* `actionType` - {actions}l-testing-actions}
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-visual.adoc:341]:
```suggestion
* `actionType` - {visual-testing-actions}
```

**Review by ikalinin1 — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**